ci/terraform/modules/lambda-role-policy-reduction/policy.tf

```diff
@@ -0,0 +1,73 @@
+data "aws_iam_policy_document" "logging_policy_document" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+
+    resources = [
+      "arn:aws:logs:*:*:*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "endpoint_xray_policy" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "xray:*"
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "networking_policy_document" {
+  count = var.vpc_arn == "" ? 0 : 1
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface"
+    ]
+    resources = [
+      "*"
+    ]
+    condition {
+      test     = "ArnLikeIfExists"
+      variable = "ec2:Vpc"
+      values   = [var.vpc_arn]
+    }
+  }
+}
+
+locals {
+  policy_documents_to_attach = merge({
+    base = concat([
+      data.aws_iam_policy_document.logging_policy_document.json,
+      data.aws_iam_policy_document.endpoint_xray_policy.json,
+      ],
+    var.vpc_arn == "" ? [] : [data.aws_iam_policy_document.networking_policy_document[0].json]),
+  }, var.policy_documents_to_attach)
```

Comment:
Do the order of these policies matter? i.e. if the json object keys are not returned in the same order, will that cause AWS/terraform to fail to deploy the change (see previous conversations about changes to policy ordering messing up a deployment)?